redbot/cogs/streams/streamtypes.py

```diff
@@ -280,92 +282,101 @@ def __init__(self, **kwargs):
-                data = await r.json(encoding="utf-8")
-        if r.status == 200:
-            if not data["data"]:
-                raise OfflineStream()
-            self.name = data["data"][0]["user_name"]
-            data = data["data"][0]
-            data["game_name"] = None
-            data["followers"] = None
-            data["view_count"] = None
-            data["profile_image_url"] = None
-            data["login"] = None
-
-            game_id = data["game_id"]
-            if game_id:
-                params = {"id": game_id}
-                async with aiohttp.ClientSession() as session:
-                    async with session.get(
-                        "https://api.twitch.tv/helix/games", headers=header, params=params
-                    ) as r:
-                        game_data = await r.json(encoding="utf-8")
-                if game_data:
-                    game_data = game_data["data"][0]
-                    data["game_name"] = game_data["name"]
-            params = {"to_id": self.id}
-            async with aiohttp.ClientSession() as session:
-                async with session.get(
-                    "https://api.twitch.tv/helix/users/follows", headers=header, params=params
-                ) as r:
-                    user_data = await r.json(encoding="utf-8")
-            if user_data:
-                followers = user_data["total"]
-                data["followers"] = followers
-
-            params = {"id": self.id}
-            async with aiohttp.ClientSession() as session:
+            try:
                 async with session.get(
-                    "https://api.twitch.tv/helix/users", headers=header, params=params
-                ) as r:
-                    user_profile_data = await r.json(encoding="utf-8")
-            if user_profile_data:
-                profile_image_url = user_profile_data["data"][0]["profile_image_url"]
-                data["profile_image_url"] = profile_image_url
-                data["view_count"] = user_profile_data["data"][0]["view_count"]
-                data["login"] = user_profile_data["data"][0]["login"]
-
-            is_rerun = False
-            return self.make_embed(data), is_rerun
-        elif r.status == 400:
+                    url, headers=header, params=params, timeout=aiohttp.ClientTimeout(total=None)
+                ) as resp:
+                    remaining = resp.headers.get("Ratelimit-Remaining")
+                    if remaining:
+                        self._rate_limit_remaining = int(remaining)
+                    reset = resp.headers.get("Ratelimit-Reset")
+                    if reset:
+                        self._rate_limit_resets.add(int(reset))
+
+                    if resp.status == 429:
+                        log.info(
+                            "Ratelimited. Trying again at %s.", datetime.fromtimestamp(int(reset))
+                        )
+                        return await self.get_data(url)
```

Comment:
Probably wouldn't hurt to release the response object since we're doing all of this inside the context manager and therefore blocking the underlying connection from returning back to the pool.
```suggestion
                        resp.release()
                        return await self.get_data(url)
```